binary.go

```diff
@@ -111,16 +122,21 @@ func (w *BitsWriter) writeFullInt(in uint64, len int) error {
 func (w *BitsWriter) writeFullByte(b byte) error {
 	if w.cacheLen == 0 {
 		w.bsCache[0] = b
-		_, err := w.w.Write(w.bsCache)
-		return err
-	}
+		if _, err := w.w.Write(w.bsCache); err != nil {
```

Comment:
I'd rather create

```go
func (w *BitsWriter) write() error {
  if _, err := w.w.Write(w.bsCache); err != nil {
    return err
  }
  if w.writeCb != nil {
    w.writeCb(w.bsCache)
  }  
}
```

that gets called anytime a `write` is needed.

What do you think?

Reply:
good call! that simplified the code